### PR TITLE
wxGUI/toolbars: fix toolbar tools labels

### LINKED ISSUE #1133: [Bug] Narrow toolbar shows IDs instead of labels
**Describe the bug**
Making windows with a toolbar, such as Layer Manager or Map Display, narrow causes a small arrow to show on the right. Clicking it shows a menu with tools which had to be hidden to make the window smaller. These tools are now represented by text, not by image. All that is fine. However, this text is not a label, but something which looks like some internal ID. The issue is even bigger for translations.

**To Reproduce**

1. Go to Layer Manager (or Map Display or other windows with a toolbar).
2. Make it as narrow as possible (minimal horizontal size).
3. Click the triangle/arrow at the right side of the toolbar.
4. See the "IDs" in the menu.

**Expected behavior**

Some readable text describing the tool/action should appear.

Some programs have short text associated with toolbar buttons such as Save, Open, Zoom (usually these programs allow text to appear with or instead the toolbar icons). There are already tooltips for (hopefully) all items in the toolbar present in GRASS. These could be used too, although they seem little too long for a menu entry.

**Screenshot**

Screenshot from 2020-12-03 21-57-30

**System description (please complete the following information):**

- Operating System: Linux (likely all/most OS)
- GRASS GIS 7.9 (likely all, not a new issue)
- Python: 3.8.5                                                                   
- wxPython: 4.0.7                                                                 

**Additional context**
Seems like API misunderstanding - user visible string is sourced from something which was not meant to be visible.

### PR DESCRIPTION
**Pull request by tmszi**
Fixes #1133.

So far, I've only fixed the labels for the Map Display toolbar tools.

wxgui_mapdisplay_toolbar_tools_labels_def_exp

### CODE REVIEW

**Review by wenzeslaus — CHANGES_REQUESTED** [review]
It works for me as expected (that is without the icons for me) and the code looks fine, but I think the labels need to be little more better. I don't think they should be as long as tooltips. They should be more like button texts, but they should be more than just keywords, so probably "Zoom to region" and "Zoom to extent" or something like that (the rest look fine).

File: gui/wxpython/vdigit/toolbars.py [hunk 1]
```diff
@@ -247,97 +247,139 @@ def _toolbarData(self):
...
-                )
+                ),
             )
         if not self.tools or "undo" in self.tools or "redo" in self.tools:
             data.append((None,))
         if not self.tools or "undo" in self.tools:
-            data.append(("undo", self.icons["undo"], self.OnUndo))
+            data.append(
+                (
+                    ("undo", _("Undo previous changes")),
```
Comment (veroandreo) [inline gui/wxpython/vdigit/toolbars.py:370]: Shouldn't this be singular? i.e., it will undo one by one, no?

File: gui/wxpython/wxplot/scatter.py [hunk 2]
```diff
@@ -318,19 +318,63 @@ def _toolbarData(self):
...
+                    BaseIcons["pan"],
+                    self.parent.OnDrag,
+                ),
+                (
+                    ("zoom", _("Zoom in")),
+                    BaseIcons["zoomIn"],
+                    self.parent.OnZoom,
+                ),
+                (
+                    ("unzoom", _("Zoom back")),
```
Comment (veroandreo) [inline gui/wxpython/wxplot/scatter.py:348]: Return to previous zoom (just for consistency)

File: gui/wxpython/lmgr/toolbars.py [hunk 3]
```diff
@@ -235,10 +330,22 @@ def _toolbarData(self):
...
                 (None,),
-                ("settings", icons["settings"], self.parent.OnNvizPreferences),
-                ("help", icons["help"], self.OnHelp),
+                (
+                    ("settings", _("3D view settings")),
+                    icons["settings"],
+                    self.parent.OnNvizPreferences,
+                ),
+                (
+                    ("help", _("3D view help")),
```
Comment (veroandreo) [inline gui/wxpython/lmgr/toolbars.py:345]: Just for consistency, shouldn't this one be "Show manual" too as in all other help icons?

**Review by veroandreo — CHANGES_REQUESTED** [review]
I added some minor comments, they have to do mostly with consistency. Thanks @tmszi :)

File: gui/wxpython/vdigit/toolbars.py (hunk 1 above)
Comment (tmszi) [reply gui/wxpython/vdigit/toolbars.py]: Yes I agree. I'm copy this label text from another tool label.

File: gui/wxpython/wxplot/scatter.py (hunk 2 above)
Comment (tmszi) [reply gui/wxpython/wxplot/scatter.py]: Yes I agree.

File: gui/wxpython/lmgr/toolbars.py (hunk 3 above)
Comment (tmszi) [reply gui/wxpython/lmgr/toolbars.py]: main_win_3d_view_toolbar

File: gui/wxpython/lmgr/toolbars.py [hunk 4]
```diff
@@ -130,22 +130,22 @@ def _toolbarData(self):
...
                 ),
                 (
-                    ("addrast", _("Add raster")),
+                    ("addrast", _("Add raster map")),
                     icons["addRast"],
                     self.parent.OnAddRaster,
                 ),
                 (
-                    ("rastmisc", _("Add various raster")),
+                    ("rastmisc", _("Add various raster map")),
```
Comment (veroandreo) [inline gui/wxpython/lmgr/toolbars.py:138]: As "various" implies plural nouns afterwards, I'd use `maps`. Same below.

Comment (tmszi) [reply gui/wxpython/lmgr/toolbars.py]: Thanks, I've already fixed it.

File: gui/wxpython/lmgr/toolbars.py (hunk 3 above)
Comment (wenzeslaus) [reply gui/wxpython/lmgr/toolbars.py]: For the record, I think the long-term goal should be to move the ones for 3D somewhere else (3D tab, map display, but I don't have concrete suggestion right now.

**Comment by veroandreo** [comment]
If I shrink the screen enough, I see this: 

Screenshot_2022-01-28_15-55-06

some of the icons do not appear. I see this both with single-window and split window modes.

**Comment by tmszi** [comment]
> Just tested again, and found that if I shrink the screen enough, I see this:
> some of the icons do not appear. I see this both with single-window and split window modes.

Yes you are right, because toolbar tool e.g **Zoom out** is `wx.ITEM_CHECK` type, [for a checkable tool (such tool stays pressed after it had been toggled)](https://wxpython.org/Phoenix/docs/html/wx.ToolBar.html?highlight=toolbar#wx.ToolBar.AddTool). It is default behavior I'm not changed it with this PR. 

https://github.com/OSGeo/grass/blob/1c300f64dcf0a19c1bc885332158ca4c4714ba96/gui/wxpython/mapdisp/toolbars.py#L178

mapdisplay_zoom_out_toolbar_tool

**Comment by veroandreo** [comment]
> > Just tested again, and found that if I shrink the screen enough, I see this:
> > some of the icons do not appear. I see this both with single-window and split window modes.
> 
> Yes you are right, because toolbar tool e.g **Zoom out** is `wx.ITEM_CHECK` type, [for a checkable tool (such tool stays pressed after it had been toggled)](https://wxpython.org/Phoenix/docs/html/wx.ToolBar.html?highlight=toolbar#wx.ToolBar.AddTool). It is default behavior I'm not changed it with this PR.
> 
> https://github.com/OSGeo/grass/blob/1c300f64dcf0a19c1bc885332158ca4c4714ba96/gui/wxpython/mapdisp/toolbars.py#L178
> 
> mapdisplay_zoom_out_toolbar_tool

Good! Thanks for the explanation :) I was not sure if it was actually related or not. +1 for merging from my side

**Comment by ninsbl** [comment]
If there is an RC2 for 8.0.1, can this make it into the milestone? Or should we bump up to 8.0.2?

**Comment by neteler** [comment]
"This branch has conflicts that must be resolved"

In the first place the PR needs to be rebased.

**Comment by tmszi** [comment]
@ninsbl @neteler  Rebase + tested again from my side.

**Review by veroandreo — APPROVED** [review]

**Review by petrasovaa — APPROVED** [review]

**Review by wenzeslaus — APPROVED** [review]

**Comment by neteler** [comment]
@tmszi may this be merged and backported? We are getting close to the releases.

**Comment by neteler** [comment]
I tried to cherry-pick backport this but too many merge conflicts arise...
Apparently another PR needs to be backported, too?

**Comment by wenzeslaus** [comment]
This is a large change touching many files and many lines while not fixing any critical bugs, so let's not backport it.

**Pull request merged**